pkg/standalone/common.go

```diff
@@ -44,9 +47,94 @@ func binaryFilePath(binaryDir string, binaryFilePrefix string) string {
 }
 
 func DefaultComponentsDirPath() string {
-	return path_filepath.Join(defaultDaprDirPath(), defaultComponentsDirName)
+	return path_filepath.Join(defaultDaprDirPath(), utils.DefaultComponentsDirName)
+}
+
+func DefaultResourcesDirPath() string {
+	return path_filepath.Join(defaultDaprDirPath(), utils.DefaultResourcesDirName)
+}
+
+// GetResourcesDir returns the path to the resources directory if it exists, otherwise it returns the path of components directory.
+// TODO: Remove this function and replace all its usage with above defined `DefaultResourcesDirPath` when `--components-path` flag is removed.
+func GetResourcesDir() string {
+	defaultResourcesDirPath := DefaultResourcesDirPath()
+	if _, err := os.Stat(defaultResourcesDirPath); os.IsNotExist(err) {
+		return DefaultComponentsDirPath()
+	}
+	return defaultResourcesDirPath
 }
 
 func DefaultConfigFilePath() string {
 	return path_filepath.Join(defaultDaprDirPath(), defaultConfigFileName)
 }
+
+// copyFilesAndCreateSymlink copies files from src to dest. It deletes the existing files in dest before copying from src.
+// this method also deletes the components dir and makes it as a symlink to resources directory.
+// please see this comment for more details:https://github.com/dapr/cli/pull/1149#issuecomment-1364424345
+// TODO: Remove this function when `--components-path` flag is removed.
```

Comment:
I have a suggestion to break it into two methods. (1) moveDir(src, dest) (which will delete files in dest, copy things from src to dest, and delete src - essentially a `mv` operation) , and (2) createSymLink (which you already have). Easier to read and maintain. The caller will then invoke these two methods serially.

Also we can replace references to `resources` and `components` in these functions, only use `src` and `dest` terminologies.